File: WRUW/Shows ViewControllers/ArchiveTable/ArchiveViewController.m
#import "ArchiveViewController.h"
#import "SongTableViewCell.h"
#import <Social/Social.h>
#import "WRUWModule-Swift.h"

@interface ArchiveViewController ()
{
    Playlist *_archive;
    UIActivityIndicatorView *spinner;
}

@end

@implementation ArchiveViewController

@synthesize currentPlaylist, currentShowTitle, tableView;

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self setTitle:currentPlaylist.dateString];

    spinner = [[UIActivityIndicatorView alloc] init];
    spinner.center =
        CGPointMake(self.view.frame.size.width / 2.0,
                    self.view.frame.size.height / 2.0);
    
    [spinner setActivityIndicatorViewStyle:UIActivityIndicatorViewStyleGray];
    spinner.color = [[ThemeManager current] wruwMainOrangeColor];
    [self.view addSubview:spinner];

    [spinner startAnimating];
    
    [tableView loadWithShow:currentPlaylist.showName.asQuery
                       date:currentPlaylist.dateString];
}

@end

File: WRUW/Shows ViewControllers/ArchiveTable/ArchiveViewController.swift
import Foundation

class ArchiveViewController: UIViewController {
    var currentPlaylist: PlaylistInfo!,
        currentShowTitle: String!

    @IBOutlet var tableView: PlaylistTableView!

    override func viewDidLoad() {
        super.viewDidLoad()

        title = currentPlaylist.dateString

        let showName = (currentPlaylist.showName as NSString).asQuery as String

        tableView.load(show: showName, date: currentPlaylist.dateString)
    }
}
